### DISCUSSION TITLE
Update Progress Bar Color?

### DISCUSSION THREAD

**Discussion by SakiSakiSakiSakiSaki** [Textualize/rich#2899] [Q&A]
Is it possible to change the bar's attributes within the Progress object after it's already been defined?

For example, updating the color of a progress bar, of a task.

Or giving two tasks/bars two different colors?

**Comment by geekya215** [comment]
I guess you want to change the default color of the progress bar, see #2399 may be helpful

**Reply by SakiSakiSakiSakiSaki** [reply]
I actually do know how to define the color of the progress bars,

```py
with Progress(TextColumn('{task.description}'),
                BarColumn(bar_width=20, 
                            complete_style='white', 
                            finished_style='bold white'),
                transient = True) as progress:
```

What I'm asking about is, how do I change the color of the progress bar **again**, from within the _with statement_?

Like maybe, there's a way to change the color of an individual progress bar, among multiple ones within the same with statement?
Maybe the following:
```py
progress.update(rich_task, advance = 1, description = message)
```
has BarColumn attributes? Not sure.

To clarify, I have a use-case where when I pausing my `for loop` because of rate-limiting, I would like the bar to momentarily turn red, or maybe even disappear. When the rate-limit wait period is over, turn it back to white.
